### PR TITLE
Don't import enum in Python 2 stubs

### PR DESCRIPTION
**Pull request by JukkaL**
Python 2.7 doesn't have enum in the stdlib, in particular, so we
shouldn't import it.

It would probably be a better long-term fix to move the Python 2 stubs
to `@python2` and simplify them there.

### CODE REVIEW

File: stdlib/inspect.pyi [hunk 1]
```diff
@@ -154,16 +154,23 @@ if sys.version_info >= (3, 10):
...
+        VAR_KEYWORD: int
+
+        if sys.version_info >= (3, 8):
+            description: str
+
+else:
+    # TODO: This actually doesn't exist on Python 2
+    _ParameterKind = int
 
 class Parameter:
```
Comment (JelleZijlstra) [inline stdlib/inspect.pyi:175]: I think Parameter and BoundArguments don't exist in Python 2 either, could you move them within the Python 3 block?

File: stdlib/inspect.pyi [hunk 2]
```diff
@@ -154,16 +154,23 @@ if sys.version_info >= (3, 10):
...
-class _ParameterKind(enum.IntEnum):
-    POSITIONAL_ONLY: int
-    POSITIONAL_OR_KEYWORD: int
-    VAR_POSITIONAL: int
-    KEYWORD_ONLY: int
-    VAR_KEYWORD: int
-
-    if sys.version_info >= (3, 8):
-        description: str
+if sys.version_info >= (3, 0):
```
Comment (srittau) [inline stdlib/inspect.pyi:157]: Nit: `enum` was introduced in Python 3.4, so it should either be `>= (3, 4)` or `>= (3,)` (since we only support 3.6+).

**Review by ilevkivskyi — APPROVED** [review]

File: stdlib/ssl.pyi [hunk 3]
```diff
@@ -1,10 +1,12 @@
-import enum
 import socket
 import sys
 from _typeshed import StrPath
 from typing import Any, Callable, ClassVar, Dict, Iterable, List, NamedTuple, Optional, Set, Text, Tuple, Type, Union, overload
 from typing_extensions import Literal
 
+if sys.version_info >= (3, 0):
```
Comment (srittau) [inline stdlib/ssl.pyi:7]: See above.

File: stdlib/inspect.pyi (hunk 1 above)
Comment (JukkaL) [reply stdlib/inspect.pyi]: Yeah. I moved some Python 3 only definitions behind version checks.

File: stdlib/inspect.pyi (hunk 2 above)
Comment (JukkaL) [reply stdlib/inspect.pyi]: Updated.

File: stdlib/ssl.pyi (hunk 3 above)
Comment (JukkaL) [reply stdlib/ssl.pyi]: Updated.

**Review by JelleZijlstra — APPROVED** [review]

**Comment by JukkaL** [comment]
Oops, actually we already have Python 2 specific stubs for inspect. I'm going to revert the changes to `inspect`.

**Review by srittau — APPROVED** [review]

**Pull request merged**